Added mailroom_part2.py

RimleeTalukdar44 opened the pull request:
Consider mailroom_part2.py
Please ignore mailroom2.py. Its a duplicate copy of mailroom_part2.py. I could not remove the file from Staging.

RimleeTalukdar44 committed:
Added mailroom_part2.py

RimleeTalukdar44 committed:
Merge branch 'master' of https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom

nairraghav commented:
You should be able to run: git rm mailroom2.py (that will remove it upon your next commit/push)

nairraghav commented on students/RimleeT/lesson04/mailroom_part2.py:
```diff
@@ -0,0 +1,120 @@
+
+
+def do_sendlettertoalldonors():
+    dir_name = tempfile.mkdtemp(prefix = 'Letters-', dir = "C:/Users/rimle/SP_2019_210A_classroom/students/RimleeT/lesson04/")
```
Be careful with setting a specific dir here, when I run this from my machine it throws an error. You can just create the files within this directory itself for now. (No need to specify the path)

nairraghav commented:
Since this is the only file for this PR, I will not merge it yet. Please address the issue and use git rm to remove the duplicate file.

RimleeTalukdar44 committed:
Merge branch 'master' of https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom

RimleeTalukdar44 committed:
Merge branch 'master' of https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom

RimleeTalukdar44 committed:
Modified mailroom_part2.py

RimleeTalukdar44 committed:
Added dict_lab.py

RimleeTalukdar44 committed:
Added file_lab.py

RimleeTalukdar44 committed:
Added trigram.py

RimleeTalukdar44 commented:
Submitting lesson04 exercises-
1. Modified mailroom_part2.py
    Changed directory path to default directory.
2. Removed duplicate mailroom2.py
3. Added dict_lab.py
4. Added file_lab.py
5. Added trigram.py

RimleeTalukdar44 committed:
Added comprehensions_lab.py

RimleeTalukdar44 committed:
Added except_exercise.py

RimleeTalukdar44 committed:
Added exception_lab.py

RimleeTalukdar44 committed:
Added mailroom3.py

RimleeTalukdar44 committed:
Added colors.py

RimleeTalukdar44 committed:
Added caught_speeding.py

RimleeTalukdar44 committed:
Added walnut_party.py

RimleeTalukdar44 committed:
Added test_walnut_party.py

RimleeTalukdar44 committed:
Added test_caught_speeding.py

RimleeTalukdar44 committed:
Added test_colors.py

RimleeTalukdar44 commented:
Submitting lesson05 exercises-
1. Added mailroom3.py
2. Added exception_lab.py
3. Added except_exercise.py
4. Added comprehensions_lab.py

RimleeTalukdar44 commented:
Submitting lesson06 exercises-
1. Added colors.py
2. Added test_colors.py
3. Added caught_speeding.py
4. Added test_caught_speeding.py
5. Added walnut_party.py
6. Added test_walnut_party.py

RimleeTalukdar44 committed:
Added mailroom4.py

RimleeTalukdar44 committed:
Added test_mailroom4.py

nairraghav commented on students/RimleeT/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,95 @@
+print('s2: ', s2)
+print('union: ', s1.union(s2))
+print('intersection: ', s1.intersection(s2))
+
```
Nice job!

nairraghav commented on students/RimleeT/lesson04/file_lab.py:
```diff
@@ -0,0 +1,47 @@
+What might be the best data structure to use to keep track of bunch of values (the languages) without duplication?"""
+
+list_of_languages = set()
+with open('D://students.txt','r') as f:
```
Be careful with hard coding locations!

nairraghav commented on students/RimleeT/lesson04/file_lab.py:
```diff
@@ -0,0 +1,47 @@
+        line_updated = line.strip().split(':')
+        if line_updated[1]:
+            nickname_languages = line_updated[1]
+            nickname_languages_updated = nickname_languages.strip().split(',')
```
instead of creating an updated variable here, you can do it above:
nickname_languages = line_udpated[1].strip().split(',')

nairraghav commented on students/RimleeT/lesson04/file_lab.py:
```diff
@@ -0,0 +1,47 @@
+        if line_updated[1]:
+            nickname_languages = line_updated[1]
+            nickname_languages_updated = nickname_languages.strip().split(',')
+            if nickname_languages_updated:
```
This will always exist. Even if it doesn't have a comma, you will always have something

nairraghav commented on students/RimleeT/lesson04/file_lab.py:
```diff
@@ -0,0 +1,47 @@
+            if nickname_languages_updated:
+                for nick_lang_list in nickname_languages_updated:
+                    nick_lang_list = nick_lang_list.strip()
+                    if nick_lang_list:
```
Same goes for here, you don't need to check if it exists since it always will.

nairraghav commented on students/RimleeT/lesson04/mailroom_part2.py:
```diff
@@ -0,0 +1,122 @@
+        
+        option()
+
+if __name__ == "__main__":
```
Nice job!

nairraghav commented on students/RimleeT/lesson04/trigram.py:
```diff
@@ -0,0 +1,47 @@
+        
+    return trigram
+
+def write_story(trigram):
```
You have the logic for building/printing the trigrams, nice job! It would be good if you could also implement reading from the sherlock.txt to build the trigrams. But that should be a minimal change. I will give you the credit for it but, if you have some time, try to implement it!

nairraghav commented on students/RimleeT/lesson05/except_exercise.py:
```diff
@@ -0,0 +1,55 @@
+    except:
+        pass
+finally:
+    last_fun()
```
Looks good!

nairraghav commented on students/RimleeT/lesson05/mailroom3.py:
```diff
@@ -0,0 +1,125 @@
+def get_donation_amount():
+    try:
+        return float(input("Enter the donation amount: "))
+    except:
```
This will catch your error but we usually want to avoid general except catchers. We want to catch specific ones so that we are aware of the specific error we see. Otherwise you could be catching errors that are not expected and run into others issues

nairraghav commented:
Nice job Rimlee! Please take a look at Mailroom3 and try to add more specifics to your excepts there. You don't want to throw a generic except because it will catch all errors, not the specific one you are looking for.
